mentat/code_file_manager.py

```diff
@@ -60,140 +56,54 @@ def _print_path_tree(tree, changed_files, cur_path, prefix=""):
             _print_path_tree(tree[key], changed_files, cur, new_prefix)
 
 
-def _is_file_text_encoded(file_path):
-    try:
-        # The ultimate filetype test
-        with open(file_path) as f:
-            f.read()
-        return True
-    except UnicodeDecodeError:
-        return False
-
-
-def _abs_files_from_list(paths: Iterable[str], check_for_text: bool = True):
-    files_direct = set()
-    file_paths_from_dirs = set()
-    for path in paths:
-        file = CodeFile(path)
-        path = Path(file.path)
-        if path.is_file():
-            if check_for_text and not _is_file_text_encoded(path):
-                logging.info(f"File path {path} is not text encoded.")
-                raise UserError(f"File path {path} is not text encoded.")
-            files_direct.add(file)
-        elif path.is_dir():
-            nonignored_files = set(
-                map(
-                    lambda f: os.path.realpath(path / f),
-                    get_non_gitignored_files(path),
-                )
-            )
-
-            file_paths_from_dirs.update(
-                filter(
-                    lambda f: (not check_for_text) or _is_file_text_encoded(f),
-                    nonignored_files,
-                )
-            )
-
-    files_from_dirs = [CodeFile(path) for path in file_paths_from_dirs]
-    return files_direct, files_from_dirs
-
-
-def _abs_file_paths_from_list(paths: Iterable[str], check_for_text: bool = True):
-    files_direct, files_from_dirs = _abs_files_from_list(paths, check_for_text)
-    return set(map(lambda f: f.path, files_direct)), set(
-        map(lambda f: f.path, files_from_dirs)
-    )
-
-
 class CodeFileManager:
     def __init__(
         self,
-        paths: Iterable[str],
-        exclude_paths: Iterable[str],
         user_input_manager: UserInputManager,
         config: ConfigManager,
-        git_root: str,
+        code_context: CodeContext,
     ):
-        self.config = config
-        self.git_root = Path(git_root)
-        self._set_file_paths(paths, exclude_paths)
         self.user_input_manager = user_input_manager
+        self.config = config
+        self.code_context = code_context
 
-        if self.files:
+        if self.code_context.files:
             cprint("Files included in context:", "green")
         else:
             cprint("No files included in context.\n", "red")
             cprint("Git project: ", "green", end="")
-        cprint(self.git_root.name, "blue")
+        cprint(self.config.git_root.name, "blue")
         _print_path_tree(
-            _build_path_tree(self.files, self.git_root),
-            get_paths_with_git_diffs(self.git_root),
-            self.git_root,
+            _build_path_tree(self.code_context.files.values(), self.config.git_root),
+            get_paths_with_git_diffs(self.config.git_root),
+            self.config.git_root,
         )
         print()
```

Comment:
I think displaying the project tree / include files should be done by CodeContext. And instead of being in the __init__ should be in something like CodeContext.display_context_tree(), called in app.py after initializing the CodeContext.